src/OrchardCore/OrchardCore.Infrastructure/Documents/DocumentManager.cs

```diff
@@ -364,23 +364,20 @@ protected async Task InvalidateInternalAsync(TDocument document, bool failover =
 
         private async Task<TDocument> GetFromDistributedCacheAsync()
         {
-            byte[] data = null;
-
             if (_isDistributed)
             {
-                data = await _distributedCache.GetAsync(_options.CacheKey);
+                byte[] data = await _distributedCache.GetAsync(_options.CacheKey);
+                if (data == null)
+                {
+                    return null;
+                }
+                return await _options.Serializer.DeserializeAsync<TDocument>(data);
             }
             else if (_memoryCache.TryGetValue<TDocument>(_options.CacheKey, out var cached))
             {
-                data = await _options.Serializer.SerializeAsync(cached);
```

Comment:
In fact there is a reason for doing this, this only happens if not distributed and on updating, when we want to get a mutable document for updating for which we don't want to return a shared document from the memory cache.

So serializing and deserializing is a way to clone the document.

Reply:
Thank you for your reply. I didn't think of these, maybe we should add some notes to the text here